src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,47 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+
+public class Contact {
+	public final String value;
+	private boolean isPrivate;
+	
+	public Contact(String contact, boolean isPrivate, String errorMessage, String validRegex) throws IllegalValueException {
+		this.value = contact.trim();
+		this.isPrivate = isPrivate;
+		if(!isValidContact(validRegex)){
+			throw new IllegalValueException(errorMessage);
+		}
+	}
+	
+	/**
+	 * returns true if this.value matches regex
+	 * @param regex
+	 * @return
+	 */
+	private boolean isValidContact(String regex){
```

Comment:
Is there anyway we can avoid passing in the regex again since it is also passed in for the constructor?